platform/linuxbsd/display_server_x11.cpp

```diff
@@ -4339,6 +4367,10 @@ DisplayServerX11::~DisplayServerX11() {
 	if (xmbstring) {
 		memfree(xmbstring);
 	}
+
+#ifdef DBUS_ENABLED
+	memfree(screensaver);
```

Comment:
```suggestion
	memdelete(screensaver);
```

Reply:
I'm happy to make this change, but can you explain what the difference is? What makes this `memdelete` a better choice than `memfree` in this instance? I'm pretty new to C++, so the answer might be really simple.

Reply:
In C++, memory allocated with `new` must be freed with `delete` and memory allocated with `alloc` must use `free`. Depending on the implementation `new` and `alloc` can track allocated memory separately (https://isocpp.org/wiki/faq/freestore-mgmt#mixing-malloc-and-delete), and `free` won't call object destructor.

With current Godot memory management implementation and no destructor in the `screensaver`, `memfree` might work, but it is wrong anyway.

Reply:
Gotcha, thanks a lot for the explanation!